test/AutoRest.TestServerLowLevel.Tests/LowLevel/Generation/ModelWriterTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections.Generic;
+using AutoRest.CSharp.Common.Input;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Models.Types;
+using NUnit.Framework;
+
+namespace AutoRest.CSharp.Generation.Writers.Tests
+{
+    public class LowLevelModelWriterTests
+    {
+        [Test]
+        public void InputBasic()
+        {
+            // refer to the original CADL file: https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-models-input-basic/sdk/template/Azure.Template/src/Generated/Models/InputModel.cs
+            var model = new ModelTypeProvider(
+                new InputModelType("InputModel", "Cadl.TestServer.InputBasic", "public",
+                    new List<InputModelProperty>{
+                        new InputModelProperty("requiredString", "requiredString", "Required string, illustrating a reference type property.", InputPrimitiveType.String, true, true, false),
+                        new InputModelProperty("requiredInt", "requiredInt", "Required int, illustrating a value type property.", InputPrimitiveType.Int32, true, true, false)
+                    },
+                    null, null, null),
+                new TypeFactory(null),
+                "test",
+                null);
+
+            var codeWriter = new CodeWriter();
+            LowLevelModelWriter.WriteModel(codeWriter, model);
+            var codes = codeWriter.ToString();
+
+            Assert.AreEqual(codes, @"// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
+
+#nullable disable
+
+using System;
+using Azure.Core;
+
+namespace Cadl.TestServer.InputBasic
+{
+public partial class InputModel
```

Comment:
Looks there are some difference compared with https://github.com/annelo-msft/azure-sdk-for-net/blob/cadl-models-input-basic/sdk/template/Azure.Template/src/Generated/Models/InputModel.cs,  is it expected?

Reply:
@lirenhe Yes, because this is the raw output from `CodeWriter`. The final codes are optimized and formatted by `Roslyn`. I consider validating the raw output because:
1. It's a unit test
2. It's not difficult to compare with the final output
3. It's cumbersome to set up `Roslyn` to process the codes

Reply:
I see.  Do we need to switch the order of properties and constructor based on the above sample?

Reply:
@lirenhe That sequence is in the original `ModelWriter`, and I didn't change it. I think it's a convention to put fields declaration before constructors. See the [MS C# Code Convention](https://docs.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions).